design/space_views.md

```diff
@@ -0,0 +1,152 @@
+  * the later is doubtful since again 3D content may be displayed in 2D views
+
+
+Note that the strong overlaps between 2D and 3D views constrain re-usability points in the design
+of `SpaceViewClass`
+
+
+### Space View State
+In addition to blueprint stored data, a space view has a class specific `SpaceViewState`
+which stored ephemeral state that is not persisted as part of the blueprint.
+This is typically used for animation/transition state.
+
+⚠️ As of writing, we're using this also for state that *should* be be persisted and needs to be moved to
+blueprint components.
+
+
+### Scene
+Scenes are created every frame from the data blueprint in order to draw the view.
+Each Space View Class defines which scene parts and scene context parts it needs to create its scene.
+
+#### `ScenePart`
+A scene part defines how a **single given archetype** is processed during the scene buildup.
+As part of its population it may set arbitrary temporary internal state and emit re_renderer drawables.
+
+#### `SceneContextPart`
+Similar to `ScenePart` but does not emit drawables. Accessible while scene parts are populated.
+This is used e.g. to prepare the transform tree.
+
+#### Scene Lifecycle
+Each frame, each `SpaceViewClass` builds up a scene. The framework defines a fixed lifecycle for all views.
+Given a `SpaceViewClass` `MyClass`:
+* default instantiate a new `TypedScene<MyClass>`
+  * this contains an instance of ``MyClass::SceneParts` and `MyClass::SceneParts::Context`,
+    each of which are collection of default initialized scene parts and contexts respectively
+* `MyClass::prepare_populate()`
+* `TypedScene<MyClass>::populate()`
+  * for each `SceneContextPart` call `populate` (can be parallelized in the future!)
+  * for each `ScenePart` call `populate`, passing in all contexts (can be parallelized in the future!)
+* `SpaceViewClass::ui()`, passing in the now populated `TypedScene<MyClass>` as well as a stored instance of `MyClass::State`
+
+
+### Space View Class Registry
+Despite being few in numbers, Space Views Classes are registered on startup.
+This is desirable since:
+* forces decoupling from other aspects of the Viewer (Viewer should be composable)
+* allows for user defined space views
+
+
+#### User defined Space View Classes
+Rust developers can use the Class Registry to register their own Space View types.
+We do *not* expect this to be a common workflow, but more of a last resort / highest level
+extensibility hooks.
+
+These user defined Space Views have no limitations over built-in Space Views and are able
+to completely reimplement existing Space Views if desired.
+
+A more common extension point in the future will be extension of the Spatial Space View Classes
+by adding new `ScenePart` to them.
+
+
+
+## Rename suggestions
+* `Scene` -> `SpaceViewFrame`
+* `SceneContext` -> `SpaceViewFrameContext`
+* `ScenePartCollection` -> `SpaceViewRenderer`
```

Comment:
How does this functionally differ from `SpaceViewFrame`?   Right now we don't have trait impls of `Scene`, but that could be a different way to organize it where we introduce, e.g. `SpatialSpaceViewFrame`.

Reply:
Or said another way: "How does a ScenePartCollection differ from a Scene?"

Reply:
What's very confusing in the current/upcoming design is that a Scene is Context + Collection but then the Collection defines the Context type. Need to rework this. A comment further up might also be resolved by that.